src/components/ui/expandable-text.tsx

```
@@ -25,81 +44,62 @@ function ExpandableTextContent({
+  const { isExpanded, shouldTruncate, setShouldTruncate } = context;
+
+  if (typeof children !== "string") {
+    // Optionally, you could warn in development mode
+    // if (process.env.NODE_ENV === "development") {
+    //   console.warn("ExpandableTextContent expects children to be a string.");
+    // }
+    return null;
+  }
+  const text = children;
+  // Check if text should be truncated using useEffect to avoid state update during render
   useEffect(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)